src/core/meshcop/dtls.cpp

```diff
@@ -141,10 +175,30 @@ otError Dtls::Start(bool             aClient,
                                        MBEDTLS_SSL_TRANSPORT_DATAGRAM, MBEDTLS_SSL_PRESET_DEFAULT);
     VerifyOrExit(rval == 0);
 
+#if OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE
+    if (mVerifyPeerCertificate)
+    {
+        mbedtls_ssl_conf_authmode(&mConf, MBEDTLS_SSL_VERIFY_REQUIRED);
+    }
+    else
+    {
+        mbedtls_ssl_conf_authmode(&mConf, MBEDTLS_SSL_VERIFY_NONE);
+    }
+#endif // OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE
+
     mbedtls_ssl_conf_rng(&mConf, mbedtls_ctr_drbg_random, &mCtrDrbg);
     mbedtls_ssl_conf_min_version(&mConf, MBEDTLS_SSL_MAJOR_VERSION_3, MBEDTLS_SSL_MINOR_VERSION_3);
     mbedtls_ssl_conf_max_version(&mConf, MBEDTLS_SSL_MAJOR_VERSION_3, MBEDTLS_SSL_MINOR_VERSION_3);
-    mbedtls_ssl_conf_ciphersuites(&mConf, ciphersuites);
+    if (!mApplicationCoapSecure)
+    {
+        mbedtls_ssl_conf_ciphersuites(&mConf, ciphersuites);
+    }
+#if OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE
```

Comment:
avoid this macro in this class? How about replace `ciphersuites` with `mCiphreSuite` and initialize `mCiphreSuite` in the constructor.

Reply:
* Ok. I have removed `OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE` macros in `dtls.cpp` and `dtls.hpp`.
* `ciphersuites` renamed to `mCipherSuites` and shared between meshcop and Application CoAPS.

Done